airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -583,6 +584,77 @@ def get_pipeline_run(
         """
         return self.get_conn().pipeline_runs.get(resource_group_name, factory_name, run_id, **config)
 
+    def check_pipeline_run_status(
```

Comment:
This function does more than it's name.
It check for status + allows you to wait for status but it doesn't let you do that to any status you want.
This function is very binded to the operator/sensor you added and does't give much flexibility for custom usages.

WDYT about splitting it to:
`check_pipeline_run_status` -  just check if run_id status match any of provided status list.
`wait_pipeline_run_termination` - wait for pipeline to end in any of the terminated states.

Reply:
Much better implementation; these functions should only perform one actions. I'll go ahead and make these changes.

Reply:
For now I chose to have two functions: 
`get_pipeline_run_status` - Simply retrieves the current status of the pipeline run
`wait_for_pipeline_run_status` - Periodically checks if the pipeline run status matches a terminal status or any input status(es)

After iterating a few times, splitting the functionality like we thought always led to instances of having unnecessary extra connections to ADF. (e.g. at the end of run a connection is made to ADF to wait for termination (via `wait_for_pipeline_run_termination`, pipeline reaches a terminal status, and then connect again to check if the status was expected (via 'check_pipeline_run_status`) even though the logic had just done this task. Felt like the solution too isolated functionally. The `wait_for_pipeline_run_status` function seems aligned with some other similar implementations but let me know what you think.